### PR TITLE
Feature  catassign  v1

### PR DESCRIPTION
**Pull request by jackholt44**
**### Category updates**
- Updates to routes to make dynamic
- Removal of Plaid Categories in transaction state
- Removal of categoryName from transaction state
- Addition of relation to Category table for categoryId
- Updates now made directly in controller not service
- Updated modal to make selection easier on long lists - done

note: also some small LnF updates to transaction list - including adding Item name
note: i have realised I could have kept the list item as SFC and passed the bank name as a separate react component - I will make this change when I get the chance

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,14 +0,0 @@
-SERVER_PORT=8130
```
Comment (atreidesend) [inline .env.example]: why deleted?

File: client/actions/TransactionActions.ts [hunk 2]
```diff
@@ -18,6 +21,31 @@ export const setTransactions = createAction<
...
+  'OPEN_CATEGORIZE_TRANSACTION_MODAL'
+)
+
+export const closeCategorizeTransactionModal = createAction(
+  'CLOSE_CATEGORIZE_TRANSACTION_MODAL'
+)
+
+export const categorizeTransactionParentIdSelected = createAction<
+  number | null
+>('categorize_TRANSACTION_PARENT_ID_SELECTED')
```
Comment (atreidesend) [inline client/actions/TransactionActions.ts]: caps

File: client/actions/TransactionActions.ts [hunk 3]
```diff
@@ -70,9 +98,11 @@ export const submitAllocateTransaction = (
   params: AllocateTransactionParams
 ): ThunkAction => async (dispatch, getState) => {
   const user = selectUser(getState())
-  if (!user) throw new UserRequiredError('User required to update items')
+  if (!user) {
```
Comment (atreidesend) [inline client/actions/TransactionActions.ts]: nice, thx for cleaning these

File: client/actions/TransactionActions.ts [hunk 4]
```diff
@@ -88,3 +118,27 @@ export const submitAllocateTransaction = (
...
+    throw new UserRequiredError('User required to categorize transactions')
+  }
+  if (!user.groupId) {
+    throw new GroupRequiredError('Group required to categorize transactions')
+  }
+
+  dispatch(submitCategorizeTransactionRequest())
+  const transactionApi = new TransactionResource(getState(), dispatch)
+  const resp = await transactionApi.CategorizeTransaction(params)
```
Comment (atreidesend) [inline client/actions/TransactionActions.ts]: `CategorizeTransaction`-> `categorizeTransaction` -> methods should be camelCase, classes / interfaces are PascalCase

File: client/api/resources/transaction.ts [hunk 5]
```diff
@@ -32,4 +34,17 @@ export default class TransactionResource extends ApiResource {
       return this.handleError(err)
     }
   }
+
+  async CategorizeTransaction(params: CategorizeTransactionParams) {
```
Comment (atreidesend) [inline client/api/resources/transaction.ts]: `categorizeTransaction`

File: client/api/resources/transaction.ts [hunk 6]
```diff
@@ -32,4 +34,17 @@ export default class TransactionResource extends ApiResource {
...
+    try {
+      const resp = await this.req.put<CategorizeTransactionResponseBody>(
+        `/transaction/${params.transaction.id}/updatecategory/`,
```
Comment (atreidesend) [inline client/api/resources/transaction.ts]: no trailing `/`

File: client/components/category/List.tsx [hunk 7]
```diff
@@ -13,14 +13,20 @@ interface StateProps {
...
+    {categoryIds.map(id => (
+      <CategoryListItem key={id} categoryId={id} />
+    ))}
   </div>
 )
 
-export const RootCategoryList = connect((state: State): StateProps => ({
-  categoryIds: selectRootCategoryIds(state),
-}))(CategoryList)
+export const RootCategoryList = connect(
```
Comment (atreidesend) [inline client/components/category/List.tsx]: let's call this `ParentCategoriesList` for consistency with the selector name

File: .env.example (hunk 1 above)
Comment (jackholt44) [reply .env.example]: Deleted in error

File: client/actions/TransactionActions.ts (hunk 4 above)
Comment (jackholt44) [reply client/actions/TransactionActions.ts]: done - find/replace error!

File: .env.example (hunk 1 above)
Comment (jackholt44) [reply .env.example]: assuming there is no point me re-adding

Comment (atreidesend) [reply .env.example]: no please re-add, or undo this commit (prob easier to re-add), important to have the example file there for future dev

Comment (jackholt44) [reply .env.example]: done

File: .env.example [hunk 8]
```diff
@@ -11,4 +11,4 @@ PLAID_API_KEY=abc1234
 PLAID_API_SECRET=abc123
 PLAID_API_URL=https://sandbox.plaid.com
 PLAID_API_ENV=sandbox
-PLAID_WEBHOOK_BASE_URL=https://yourlocal.dev.tunnel.here.dev/api/webhook
+PLAID_WEBHOOK_BASE_URL=https://yourlocal.dev.tunnel.here.dev/api/webhook
```
Comment (atreidesend) [inline .env.example]: set this in vscode:
image

Comment (atreidesend) [reply .env.example]: not a blocker, just good to be consistent

Comment (atreidesend) [reply .env.example]: actually, set both:
image

File: client/components/transaction/TransactionItem.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+type Props = OwnProps & StateProps
+
+class TransactionItem extends React.PureComponent<Props> {
+  render() {
+    const { transaction } = this.props
+    return (
+      <div className="transaction-list-item__bank">
+        {transaction.itemId ? this.findItemName() : 'Set Category'}
```
Comment (atreidesend) [inline client/components/transaction/TransactionItem.tsx]: what is this? if it has no item, set category?

File: client/components/transaction/TransactionItem.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+interface StateProps {
+  categories: ReadonlyArray<CategoryForClient>
+  items: ReadonlyArray<ItemForClient>
+}
+
+type Props = OwnProps & StateProps
+
+class TransactionItem extends React.PureComponent<Props> {
```
Comment (atreidesend) [inline client/components/transaction/TransactionItem.tsx]: what's the purpose of this component?

**Pull request merged**